images/openwisp_base/Dockerfile

```diff
@@ -7,67 +7,78 @@ FROM python:3.10-slim-bullseye AS SYSTEM
-                python3-pip libxml2-dev libxslt1-dev zlib1g-dev g++
+    gcc libpq-dev libjpeg-dev libffi-dev python3-dev \
+    python3-pip libxml2-dev libxslt1-dev zlib1g-dev g++
 
 RUN useradd --system --password '' --create-home --shell /bin/bash \
-   --gid root --uid 1001 openwisp
+    --gid root --uid 1001 openwisp
 USER openwisp:root
 
 FROM SYSTEM AS PYTHON
 
 ENV PATH="${PATH}:/home/openwisp/.local/bin"
 ENV PYTHONPATH=/home/openwisp/.local/lib/python3.10/site-packages
+ENV EMAIL_BACKEND=djcelery_email.backends.CeleryEmailBackend
 
-RUN pip install --user --upgrade pip
-ARG OPENWISP_MONITORING_SOURCE=https://github.com/openwisp/openwisp-monitoring/tarball/docker-openwisp-usage
+RUN pip install --user --upgrade pip~=22.0.4
+ARG OPENWISP_MONITORING_SOURCE=openwisp-monitoring~=1.0.0
+# hadolint ignore=DL3013
 RUN pip install --user --upgrade ${OPENWISP_MONITORING_SOURCE}
-ARG OPENWISP_FIRMWARE_SOURCE=https://github.com/openwisp/openwisp-firmware-upgrader/tarball/docker-openwisp-usage
+ARG OPENWISP_FIRMWARE_SOURCE=openwisp-firmware-upgrader~=1.0.0
+# hadolint ignore=DL3013
 RUN pip install --user --upgrade ${OPENWISP_FIRMWARE_SOURCE}
-ARG OPENWISP_TOPOLOGY_SOURCE=https://github.com/openwisp/openwisp-network-topology/tarball/docker-openwisp-usage
+ARG OPENWISP_TOPOLOGY_SOURCE=openwisp-network-topology~=1.0.0
+# hadolint ignore=DL3013
 RUN pip install --user --upgrade ${OPENWISP_TOPOLOGY_SOURCE}
-ARG OPENWISP_RADIUS_SOURCE=https://github.com/openwisp/openwisp-radius/tarball/docker-openwisp-usage
+ARG OPENWISP_RADIUS_SOURCE=openwisp-radius~=1.0.0
+# hadolint ignore=DL3013
 RUN pip install --user --upgrade ${OPENWISP_RADIUS_SOURCE}
 
 # here we try to install custom versions of the modules only if the
 # supplied argument does not equal the default value, because
 # otherwise these modules will have already been installed above
 ARG OPENWISP_IPAM_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$OPENWISP_IPAM_SOURCE" != "default" ] ; then \
     pip install --user --upgrade ${OPENWISP_IPAM_SOURCE}; \
     fi
 ARG OPENWISP_CONTROLLER_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$OPENWISP_CONTROLLER_SOURCE" != "default" ] ; then \
     pip install --user --upgrade ${OPENWISP_CONTROLLER_SOURCE}; \
     fi
 ARG OPENWISP_NOTIFICATION_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$OPENWISP_NOTIFICATION_SOURCE" != "default" ] ; then \
     pip install --user --upgrade ${OPENWISP_NOTIFICATION_SOURCE}; \
     fi
 ARG OPENWISP_USERS_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$OPENWISP_USERS_SOURCE" != "default" ] ; then \
     pip install --user --upgrade --force-reinstall ${OPENWISP_USERS_SOURCE}; \
     fi
 ARG OPENWISP_UTILS_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$OPENWISP_UTILS_SOURCE" != "default" ] ; then \
     pip install --user --upgrade --force-reinstall ${OPENWISP_UTILS_SOURCE}; \
     fi
 ARG DJANGO_X509_SOURCE=default
+# hadolint ignore=DL3013
 RUN if [ "$DJANGO_X509_SOURCE" != "default" ]; then \
```

Comment:
Minor nitpick, can we add some space after if, like instead of:
```
if [ "$DJANGO_X509_SOURCE" != "default" ]; then \
pip install --user --upgrade --force-reinstall ${DJANGO_X509_SOURCE}~=1.1.0; \
fi
```

Can we do:

```
if [ "$DJANGO_X509_SOURCE" != "default" ]; then \
    pip install --user --upgrade --force-reinstall ${DJANGO_X509_SOURCE}~=1.1.0; \
fi
```